core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -521,16 +509,17 @@ private[spark] class BlockManager(
               }
             }
           }
+        } else {
+          releaseLock(blockId)
+          None
         }
-      }
-    } else {
-      logDebug(s"Block $blockId not registered locally")
     }
-    None
   }
 
   /**
    * Get block from remote block managers.
+   *
+   * This does not acquire a local lock on this block.
```

Comment:
I think "This does not acquire a lock on this block in this JVM" is clearer. The concept of "local lock" is not super intuitive (is there a "remote lock"?)